### PR TITLE
Table material

### PR DESCRIPTION
**Pull request by MaxArnaut**

### CODE REVIEW

File: packages/elements-material-ui/src/Table/Table.tsx [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+  if (props.expandable) {
+    tableHooks.push(useExpanded);
+  }
+  if (props.pagination) {
+    tableHooks.push(usePagination);
+  }
+  if (props.selection) {
+    tableHooks.push(useRowSelect);
+  }
+  // tableHooks.push(useFlexLayout);
```
Comment (frontegg-david) [inline packages/elements-material-ui/src/Table/Table.tsx:110]: Why you commented this line

File: packages/elements-material-ui/src/Table/TableFilterColumn.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+              <FilterListIcon className={classes.filterIcon} key={1} name='filters' />
+            </IconButton>
+          </Tooltip>
+        }
+      />
+    </Box>
+  );
+};
+
+function useDebounce(value: any, delay: number) {
```
Comment (frontegg-david) [inline packages/elements-material-ui/src/Table/TableFilterColumn.tsx:56]: move this to core helpers and use it from @frontegg/react-core

File: packages/core/src/elements/Table/interfaces.ts [hunk 3]
```diff
@@ -38,6 +38,7 @@ export interface TableProps<T extends object = {}> {
   tableHeader?: boolean;
 
   data: T[];
+  // totalData: number;
```
Comment (frontegg-david) [inline packages/core/src/elements/Table/interfaces.ts:41]: what about totalData?  for auto pagination

**Review by frontegg-david — CHANGES_REQUESTED** [review]

File: packages/core/src/elements/Table/interfaces.ts (hunk 3 above)
Comment (MaxArnaut) [reply packages/core/src/elements/Table/interfaces.ts:41]: Do you mean the infinity scrolling?

Comment (frontegg-david) [reply packages/core/src/elements/Table/interfaces.ts:41]: No. Never mind. I need to fix something in FeTable before

**Review by frontegg-david — APPROVED** [review]

File: packages/demo-saas/src/TableExample.tsx [hunk 4]
```diff
@@ -1149,7 +1149,7 @@ export const TableExample: FC = () => {
   }, []);
 
   return (
-    <div style={{ textAlign: 'center' }}>
+    <div style={{ textAlign: 'center', paddingBottom: '20px' }}>
```
Comment (frontegg-david) [inline packages/demo-saas/src/TableExample.tsx:1152]: why exactly 20px? it this pagination height?

File: packages/elements-material-ui/src/Table/TableHead.tsx [hunk 5]
```diff
@@ -9,6 +9,12 @@ const useStyles = makeStyles((theme) => ({
   checkBox: {
     margin: '-9px 0',
   },
+  head: {
+    position: 'sticky',
+    top: '0px',
+    zIndex: 1,
+    background: theme.palette.background.paper,
```
Comment (frontegg-david) [inline packages/elements-material-ui/src/Table/TableHead.tsx:16]: Good job

File: packages/demo-saas/src/TableExample.tsx (hunk 4 above)
Comment (MaxArnaut) [reply packages/demo-saas/src/TableExample.tsx:1152]: it's just padding for div container in the example, it's not related to the table

File: packages/elements-material-ui/src/Table/TableHead.tsx (hunk 5 above)
Comment (MaxArnaut) [reply packages/elements-material-ui/src/Table/TableHead.tsx:16]: Thanks!

**Pull request merged**